app/src/main/java/com/example/lexis/fragments/PracticeFlashcardFragment.java

```diff
@@ -0,0 +1,273 @@
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        binding = FragmentPracticeFlashcardBinding.inflate(inflater);
+        return binding.getRoot();
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        if (words == null) {
+            words = new ArrayList<>();
+            fetchVocabulary(targetLanguage);
+        }
+
+        setUpCardStackView();
+        setUpButtons();
+        setUpToolbar();
+    }
+
+    /*
+    Fetch the vocabulary for the provided language and add to flashcards adapter.
+    */
+    private void fetchVocabulary(String targetLanguage) {
+        ParseQuery<Word> query = ParseQuery.getQuery(Word.class);
+        query.include(Word.KEY_USER);
+        query.whereEqualTo(Word.KEY_USER, ParseUser.getCurrentUser());
+        query.whereEqualTo(Word.KEY_TARGET_LANGUAGE, targetLanguage);
+        query.addDescendingOrder("createdAt");
+        query.findInBackground((words, e) -> {
+            if (e != null) {
+                Log.e(TAG, "Issue with getting vocabulary", e);
+                return;
+            }
+            adapter.addAll(words);
+        });
+    }
+
+    /*
+    Set up the card stack view to display the flashcards.
+    */
+    private void setUpCardStackView() {
+        adapter = new FlashcardsAdapter(this, words, answerInEnglish);
+        cardLayoutManager = new CardStackLayoutManager(getActivity(), this);
+        cardLayoutManager.setStackFrom(StackFrom.TopAndRight);
+        cardLayoutManager.setTranslationInterval(12.0f);
+
+        binding.stackFlashcards.setLayoutManager(cardLayoutManager);
+        binding.stackFlashcards.setAdapter(adapter);
+    }
+
+    /*
+    Set up buttons for automatic swiping.
+    */
+    private void setUpButtons() {
+        binding.btnForgot.setRippleColor(getResources().getColor(R.color.deep_champagne));
+        binding.btnKnow.setRippleColor(getResources().getColor(R.color.light_cyan));
+
+        binding.btnForgot.setOnClickListener(v -> {
+            if (!wasFlipped) {
+                Toast.makeText(getActivity(), "You haven't flipped this card yet! 🤔", Toast.LENGTH_SHORT).show();
+                return;
+            }
+            SwipeAnimationSetting setting = new SwipeAnimationSetting.Builder()
+                    .setDirection(Direction.Left)
+                    .setDuration(Duration.Normal.duration)
+                    .setInterpolator(new AccelerateInterpolator())
+                    .build();
+            cardLayoutManager.setSwipeAnimationSetting(setting);
+            binding.stackFlashcards.swipe();
+        });
+
+        binding.btnKnow.setOnClickListener(v -> {
+            if (!wasFlipped) {
+                Toast.makeText(getActivity(), "You haven't flipped this card yet! 🤔", Toast.LENGTH_SHORT).show();
```

Comment:
if this function will be called multiple times, you can consider pulling it out into a separate helper